test/test_torrent.cpp

```diff
@@ -248,63 +260,26 @@ TORRENT_TEST(added_peers)
 
 TORRENT_TEST(torrent)
 {
-/*	{
-		remove("test_torrent_dir2/tmp1");
-		remove("test_torrent_dir2/tmp2");
-		remove("test_torrent_dir2/tmp3");
-		file_storage fs;
-		boost::int64_t file_size = 256 * 1024;
-		fs.add_file("test_torrent_dir2/tmp1", file_size);
-		fs.add_file("test_torrent_dir2/tmp2", file_size);
-		fs.add_file("test_torrent_dir2/tmp3", file_size);
-		libtorrent::create_torrent t(fs, 128 * 1024);
-		t.add_tracker("http://non-existing.com/announce");
-
-		std::vector<char> piece(128 * 1024);
-		for (int i = 0; i < int(piece.size()); ++i)
-			piece[i] = (i % 26) + 'A';
-
-		// calculate the hash for all pieces
-		sha1_hash ph = hasher(&piece[0], piece.size()).final();
-		int num = t.num_pieces();
-		TEST_CHECK(t.num_pieces() > 0);
-		for (int i = 0; i < num; ++i)
-			t.set_hash(i, ph);
-
-		std::vector<char> tmp;
-		std::back_insert_iterator<std::vector<char> > out(tmp);
-		bencode(out, t.generate());
-		error_code ec;
-		boost::shared_ptr<torrent_info> info(boost::make_shared<torrent_info>(&tmp[0], tmp.size(), boost::ref(ec), 0));
-		TEST_CHECK(info->num_pieces() > 0);
-
-		test_running_torrent(info, file_size);
-	}
-*/
-	{
-		file_storage fs;
+	int const file_size = 1024;
+	file_storage fs;
 
-		fs.add_file("test_torrent_dir2/tmp1", 1024);
-		libtorrent::create_torrent t(fs, 128 * 1024, 6);
+	fs.add_file("test_torrent_dir2/tmp1", file_size);
+	libtorrent::create_torrent t(fs, file_size, 6);
 
-		std::vector<char> piece(128 * 1024);
-		for (int i = 0; i < int(piece.size()); ++i)
-			piece[i] = (i % 26) + 'A';
+	std::vector<char> piece(file_size);
+	for (int i = 0; i < int(piece.size()); ++i)
+		piece[i] = (i % 26) + 'A';
 
-		// calculate the hash for all pieces
-		sha1_hash ph = hasher(&piece[0], piece.size()).final();
-		int num = t.num_pieces();
-		TEST_CHECK(t.num_pieces() > 0);
-		for (int i = 0; i < num; ++i)
-			t.set_hash(i, ph);
+	// calculate the hash for all pieces
+	TEST_CHECK(t.num_pieces() == 1);
+	t.set_hash(0, hasher(&piece[0], piece.size()).final());
```

Comment:
was it deliberate to only set the piece hash for the first piece? You made the torrent only have a single please. what's the rationale?

Reply:
Previously there was only one piece too. Also I thought there was a mistake, torrent with only 1KB file and a 128 KB piece to calculate hash for. The last 127 KB of the piece should be padded with zeroes, and shouldn't participate in piece hash calculation. Does this make sense?